src/graph/parser.ts

```diff
@@ -294,6 +296,39 @@ export class Graph {
                         break;
                     }
 
+                    // Size field
+                    case "width":
+                    case "height": {
+                        requiresValue()
+
+                        //search for unit
+                        let index = (value as string).search(/[A-Za-z%]/)
```

Comment:
This seems to be a very inefficient way of separating an input into the value and its unit. 
1) Regular expressions are slow
2) This seems to match any characters, rather than just accepted characters. For example, using this code a random string such as "5734ghj4978epr" would return as "value: 5734, unit: ghj4978epr". While the code would return as an invalid unit, it would also return a useless error.